builtins/web/fetch/fetch-api.cpp

```diff
@@ -96,7 +99,273 @@ bool fetch(JSContext *cx, unsigned argc, Value *vp) {
   SetReservedSlot(request_obj, static_cast<uint32_t>(Request::Slots::PendingResponseHandle),
                   PrivateValue(pending_handle));
 
-  args.rval().setObject(*response_promise);
+  rval.setObject(*response_promise);
+  return true;
+}
+
+/// https://fetch.spec.whatwg.org/#scheme-fetch
+bool fetch_blob(JSContext *cx, HandleObject request_obj, HostString method, HostString url,
+                MutableHandleValue rval) {
+
+  RootedObject response_promise(cx, JS::NewPromiseObject(cx, nullptr));
+  if (!response_promise) {
+    return false;
+  }
+
+  rval.setObject(*response_promise);
+
+  // 1. Let blobURLEntry be request's current URL's blob URL entry.
+  RootedString blob_url(cx);
+
+  // 2. If request's method is not `GET` or blobURLEntry is null, then return a network error.
+  if (std::memcmp(method.ptr.get(), "GET", method.len) != 0) {
+    return api::throw_error(cx, FetchErrors::FetchNetworkError);
```

Comment:
I suppose this is an existing issue, but I believe fetch network errors are supposed to be DOMException::raise.

Reply:
I tried changing this calls to `return DOMException::raise(cx, "The owls are not what they seem", "TypeError");` but then I'm getting bunch of test failures like this:

```
1:   NAME:    Fetching [PUT] URL.createObjectURL(blob) is KO
1:   MESSAGE: promise_rejects_js: function "function() { throw e }" threw object "TypeError: Invalid request method for scheme fetch" ("TypeError") expected instance of function "function TypeError() {
```

I'm not sure if it's possible to use `DOMException` the way the tests expects?

Reply:
That might be a deeper issue with our implementation of DOMException then. See here how `throw a TypeError` in the spec actually means throw a TypeError which is a DOM error - https://webidl.spec.whatwg.org/#exceptiondef-typeerror.

We can handle this in a follow-up of course, just wanted to mention.

Reply:
> That all said, returning a network error looks like it should return a response? https://fetch.spec.whatwg.org/#concept-network-error

Indeed! It looks like there is a discrepancy between the tests and the spec. E.g. see this function: https://github.com/web-platform-tests/wpt/blob/master/fetch/api/basic/scheme-blob.sub.any.js#L22

It only expects `scheme-fetch` to throw  a `TypeError` on failure. I've added a `network_error` function that sets the response to the object according to the spec, but throws a `TypeError` anyway to be compliant with the tests.